connectors/tests/test_sync_job_runner.py

```diff
@@ -0,0 +1,249 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import asyncio
+from unittest.mock import ANY, AsyncMock, Mock
+
+import pytest
+
+from connectors.byoc import Filter, Pipeline
+from connectors.filtering.validation import InvalidFilteringError
+from connectors.sync_job_runner import SyncJobRunner
+
+total_document_count = 100
+
+
+def create_runner(
+    source_changed=True,
+    source_available=True,
+    async_bulk_result=None,
+):
+    source_klass = Mock()
+    data_provider = Mock()
+    data_provider.tweak_bulk_options = Mock()
+    data_provider.changed = AsyncMock(return_value=source_changed)
+    data_provider.ping = AsyncMock()
+    if not source_available:
+        data_provider.ping.side_effect = Exception()
+    source_klass.return_value = data_provider
+
+    sync_job = Mock()
+    sync_job.id = "1"
+    sync_job.configuration = {}
+    sync_job.service_type = "mysql"
+    sync_job.index_name = "search-mysql"
+    sync_job.pipeline = Pipeline({})
+    sync_job.filtering = Filter()
+    sync_job.claim = AsyncMock()
+    sync_job.done = AsyncMock()
+    sync_job.fail = AsyncMock()
+    sync_job.cancel = AsyncMock()
+    sync_job.suspend = AsyncMock()
+    sync_job.reload = AsyncMock(return_value=sync_job)
+    sync_job.validate_filtering = AsyncMock()
+
+    connector = Mock()
+    connector.id = "1"
+    connector.features.sync_rules_enabled.return_value = True
+    connector.document_count = AsyncMock(return_value=total_document_count)
+    connector.sync_starts = AsyncMock()
+    connector.sync_done = AsyncMock()
+
+    elastic_server = Mock()
+    elastic_server.prepare_content_index = AsyncMock()
+    elastic_server.async_bulk = AsyncMock(return_value=async_bulk_result)
+
+    bulk_options = {}
+
+    return SyncJobRunner(
+        source_klass=source_klass,
+        sync_job=sync_job,
+        connector=connector,
+        elastic_server=elastic_server,
+        bulk_options=bulk_options,
+    )
+
+
+@pytest.mark.asyncio
+async def test_job_claim_fail(patch_logger):
```

Comment:
Same as in the other PR. Did we agree on silencing the logger during unit tests? Could make debugging harder in CIs.